virtme_ng/run.py

```diff
@@ -992,23 +1006,38 @@ def _get_virtme_net_mac_address(self, args):
         else:
             self.virtme_param["net_mac_address"] = ""
 
-    def _get_virtme_server(self, args):
-        if args.server is not None:
-            self.virtme_param["server"] = "--server " + args.server
+    def _get_virtme_console(self, args):
+        if args.console is not None and args.ssh is not None:
+            arg_fail('--console cannot be used with --ssh', show_usage=False)
+
+        if args.console is not None:
+            self.virtme_param["console"] = f"--server console --port {args.console}"
         else:
-            self.virtme_param["server"] = ""
+            self.virtme_param["console"] = ""
+
+    def _get_virtme_console_client(self, args):
+        if args.console_client is not None:
+            self.virtme_param["console_client"] = f"--client console --port {args.console_client}"
+        else:
+            self.virtme_param["console_client"] = ""
+
+    def _get_virtme_ssh(self, args):
+        if args.console is not None and args.ssh is not None:
+            arg_fail('--console cannot be used with --ssh', show_usage=False)
+
+        if args.ssh is not None and args.root is not None:
+            arg_fail('--ssh is not currently compatible with --root', show_usage=False)
```

Comment:
So, I added this check because we don't know if the user that runs vng has write access to the rootfs and in that case it'd fail to generate the ssh keys in the cache dir.

We could export/mount a separate fs using 9p for the cache dir, like we do for the guest tools, but I didn't want to add too much stuff with this PR, so I was thinking to keep it simple and do not allow `--ssh` with `--root` for now.

Later (as a separate PR) we should probably introduce a generic shared cachedir between host and guest, where we can put data generated on-the-fly, like the ssh keys, or temporary scripts generated on the host that must be executed inside the guest (see `console_script_path`), etc.

WDYT?